ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -1346,6 +1351,11 @@ static PrjFS_Result RecursivelyMarkAllChildrenAsInRoot(const char* fullDirectory
             
             dirEntry = readdir(directory);
         }
+        
+        if (directory != nullptr)
```

Comment:
Same comment here.  Since we're already checking for `nullptr` on line 1326 I don't think we need to check again here (let me know if you see a scenario I'm missing that requires this check).